tests/helper.py

```diff
@@ -4,12 +4,105 @@
 import os
 import random
 import string
+import sys
 import unittest2
 
-from mock import patch, Mock
+from distutils.version import StrictVersion
+from mock import patch, Mock, ANY
 
 import stripe
 from stripe import six
+from stripe.six.moves.urllib.request import urlopen
+from stripe.six.moves.urllib.error import HTTPError
+
+
+MOCK_MINIMUM_VERSION = '0.4.0'
+MOCK_PORT = os.environ.get('STRIPE_MOCK_PORT', 12111)
+
+
+try:
+    resp = urlopen('http://localhost:%s/' % MOCK_PORT)
+    info = resp.info()
+except HTTPError as e:
+    info = e.info()
+except Exception:
+    sys.exit("Couldn't reach stripe-mock at `localhost:%s`. Is "
+             "it running? Please see README for setup instructions." %
+             MOCK_PORT)
+
+version = info.get('Stripe-Mock-Version')
+if version != 'master' \
+        and StrictVersion(version) < StrictVersion(MOCK_MINIMUM_VERSION):
+    sys.exit("Your version of stripe-mock (%s) is too old. The minimum "
+             "version to run this test suite is %s. Please "
+             "see its repository for upgrade instructions." %
+             (version, MOCK_MINIMUM_VERSION))
+
+
+class StripeMockTestCase(unittest2.TestCase):
+    RESTORE_ATTRIBUTES = ('api_base', 'api_key', 'client_id')
+
+    def setUp(self):
+        super(StripeMockTestCase, self).setUp()
+
+        self._stripe_original_attributes = {}
+
+        for attr in self.RESTORE_ATTRIBUTES:
+            self._stripe_original_attributes[attr] = getattr(stripe, attr)
+
+        stripe.api_base = 'http://localhost:%s' % MOCK_PORT
+        stripe.api_key = 'sk_test_123'
+        stripe.client_id = 'ca_123'
+
+        self._real_request = stripe.api_requestor.APIRequestor.request
+        self._stubbed_requests = {}
+
+        self.request_patcher = patch(
+            'stripe.api_requestor.APIRequestor.request',
+            side_effect=self._patched_request,
+            autospec=True)
+        self.request_spy = self.request_patcher.start()
+
+    def tearDown(self):
+        super(StripeMockTestCase, self).tearDown()
+
+        self.request_patcher.stop()
+
+        for attr in self.RESTORE_ATTRIBUTES:
+            setattr(stripe, attr, self._stripe_original_attributes[attr])
+
+    def _patched_request(self, requestor, method, url, *args, **kwargs):
+        if (method, url) in self._stubbed_requests:
+            response_body = self._stubbed_requests.pop((method, url))
+            return response_body, stripe.api_key
+        return self._real_request(requestor, method, url, *args, **kwargs)
+
+    def stub_request(self, method, url, response_body={}):
+        self._stubbed_requests[(method, url)] = response_body
+
+    def assert_requested(self, method, url, params=ANY, headers=ANY):
```

Comment:
OOC, do you know what's up with Python's mixing of snakecase and camelcase? It's weird to see `assertEquals` mixed with `assert_requested` everywhere. I see that PEP8 recommends snakecasing, so I guess the unittest API is mostly an accident of history that's been left in place for compatibility?

Reply:
Yeah, AFAIK it's because many parts of the standard library predate PEP8 and it was decided that maintaining backwards compatibility was more important than PEP8 compliance.

I generally try to ignore this and always use snake case for my own methods, but I can rename the method to `assertRequested` if you'd rather!

Reply:
Nah, I think what you're doing is right — just too bad this crazy historical inconsistency exists :)